docs/query-help-style-guide.md

```diff
@@ -0,0 +1,202 @@
+*   If you are using an example to illustrate the solution to a problem, and the change required is minor, avoid repeating the whole example. It is preferable to either describe the change required or to include a smaller snippet of the corrected code.
+*   Clearly indicate which of the samples is an example of bad coding practice and which is recommended practice.
+
+There are several ways of including a code example in a query help file:
+
+*   Directly include the code as nested text in the `example` section, Use the `sample` block and define a `language` attribute to specify the language of the example:
+
+```       
+<example>
+<p>This example highlights poor coding practice</p>
+<sample language = "java" />
+
+Example of poor code
+
+<p>This example shows how to fix the code</p>
+<sample language = "java" />
+
+Example of correct code
+
+</example>
+```     
+
+
+*   Define the code example in a `src` file. The language is inferred from the file extension:
+
+```
+<example>
+<p>This example highlights poor coding practice</p>
+
+<sample src = "example-code-bad.java" />
+
+<p>This example shows how to fix the code</p>
+
+<sample src = "example-code-fixed.java" />
+</example>
+```
+     
+## Including references
+
+You should include one or more references to provide further information about the problem that your query is designed to find. References can be of the following types:
+
+### Books
+
+If you are citing a book, use the following format:
+
+>\<Author-initial. Surname>, _\<Book title>_ \<page/chapter etc.\>, \<Publisher\>, \<date\>.
+
+For example:
+
+>W. C. Wake, _Refactoring Workbook_, pp. 93 – 94, Addison-Wesley Professional, 2004.
+
+Note, & symbols need to be replaced by \&amp;. The symbol will be displayed correctly in the html files generated from the qhelp files.
+
+### Academic papers
+
+If you are citing an academic paper, we recommend adopting the reference style of the journal that you are citing. For example: 
+
+>S. R. Chidamber and C. F. Kemerer, _A metrics suite for object-oriented design_. IEEE Transactions on Software Engineering, 20(6):476-493, 1994.
+
+
+### Websites
+
+If you are citing a website, please use the following format, without breadcrumb trails:
+
+>\<Name of website>: \<Name of page or anchor>
+
+For example:
+
+>Java 6 API Specification: [Object.clone()](http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html#clone%28%29).
+
+### Referencing potential security weaknesses
+
+If your query checks code for a CWE weakness, you should use the `@tags` element in the query file to reference the associated CWEs, as explained [here](query-metadata-style-guide.md). When you use these tags, a link to the appropriate entry from the [MITRE.org](https://cwe.mitre.org/scoring/index.html) site will automatically appear as a reference in the qhelp file.
+
+## Query help example 
+
+The following example is a qhelp file for a query from the standard query suite for Java: 
```

Comment:
It should probably be mentioned somewhere above that code words, such as keywords and the like, should be formatted with `<code>..</code>` as exemplified below.